exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebUserConnection.java

```diff
@@ -151,7 +151,7 @@ public void sendData(RpcOutcomeListener<Ack> listener, QueryWritableBatch result
         loader.clear();
       }
     } catch (Exception e) {
-      exception = UserException.systemError(e).build(logger);
+      throw UserException.systemError(e).build(logger);
```

Comment:
I don't think we should throw an exception here. We should stick to original approach and store it but just add method `getException()` in`AbstractDisposableUserClientConnection` class, similar to `getError()`. And then do proper handling of both.